Tools/gittools/pre-commit.py

```diff
@@ -75,10 +80,10 @@ def run(self):
                 # don't check deleted files
                 continue
             (base, extension) = os.path.splitext(filepath)
-            if extension != ".py":
-                continue
-            if not self.check_file(filepath):
-                self.retcode = 1
+            if extension == ".py" and self.check_for_flake8_tag(filepath):
+                self.files_to_check_flake8.append(filepath)
```

Comment:
Why did you change this from doing the continue if it wasn't a file we're interested in?

```suggestion
            if extension != ".py":
                continue
            if not self.has_flake8_tag(filepath):
                continue
            self.files_to_check_flake8.append(filepath)
```

Reply:
Changing from a double negative to a single positive. The loop moves on in the same way. If other file format and their associated linters were to be added, the same pattern could be used.